### PR TITLE
C++11 patchset 9: move hardcoded init parameters to class definitions (part 1)

### PR DESCRIPTION
**Pull request by nerzhul**
C++11 introduced the possibility to define the default values directly in class definitions, do it on current code

This patchset cleanup src/{a*,b*,c*-> constants.h} files

### CODE REVIEW

File: src/clientiface.h [hunk 1]
```diff
@@ -388,7 +364,7 @@ class RemoteClient
 		Block is removed when GOTBLOCKS is received.
 		Value is time from sending. (not used at the moment)
 	*/
-	std::map<v3s16, float> m_blocks_sending;
+	std::map<v3s16, float> m_blocks_sending = {};
```
Comment (SmallJoker) [inline src/clientiface.h]: Aren't sets/maps defined to be empty when initialized?

Comment (sfan5) [inline src/clientiface.h]: Yes

File: src/collision.h [hunk 2]
```diff
@@ -36,37 +36,21 @@ enum CollisionType
...
-		node_p(-32768,-32768,-32768),
-		bouncy(false),
-		old_speed(0,0,0),
-		new_speed(0,0,0)
-	{}
+	CollisionType type = COLLISION_NODE;
+	v3s16 node_p = v3s16(-32768,-32768,-32768); // COLLISION_NODE
+	bool bouncy = false;
+	v3f old_speed = v3f(0.0f, 0.0f, 0.0f);
+	v3f new_speed = v3f(0.0f, 0.0f, 0.0f);
```
Comment (SmallJoker) [inline src/collision.h]: They're defined to be [`0,0,0` in the constructor without args](http://irrlicht.sourceforge.net/docu/vector3d_8h_source.html#l00026l)

File: src/clouds.h [hunk 3]
```diff
@@ -133,11 +133,11 @@ class Clouds : public scene::ISceneNode
...
 	bool m_enable_3d;
 	u32 m_seed;
-	v2f m_camera_pos;
-	v2f m_origin;
+	v2f m_camera_pos = v2f(0.0f, 0.0f);
+	v2f m_origin = v2f(0.0f, 0.0f);
 	v2f m_speed;
-	v3s16 m_camera_offset;
-	video::SColorf m_color;
+	v3s16 m_camera_offset = v3s16(0, 0, 0);
```
Comment (SmallJoker) [inline src/clouds.h]: Also set to 0 when initialized.

File: src/clientiface.h (hunk 1 above)
Comment (nerzhul) [inline src/clientiface.h]: i forgot to remove that my bad :)

File: src/collision.h (hunk 2 above)
Comment (nerzhul) [inline src/collision.h]: then i will remove all standard v3f v3s16 :)

**Comment by nerzhul** [comment]
All points fixed

File: src/collision.cpp [hunk 4]
```diff
@@ -499,7 +498,6 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 				info.type = COLLISION_NODE;
 
 			info.node_p = nearest_info.position;
-			info.bouncy = bouncy;
```
Comment (SmallJoker) [inline src/collision.cpp]: Why were these removed? `bouncy`, `collides_xz` and `standing_on_unloaded` might be used in the future, if they aren't right now.

Comment (nerzhul) [inline src/collision.cpp]: there are affected but never used since 2012 it's why i removed them from code

**Comment by nerzhul** [comment]
I added a second commit to this PR to make things advance (it's boring task but should be done)

File: src/content_sao.h [hunk 5]
```diff
@@ -368,47 +368,47 @@ class PlayerSAO : public UnitSAO
...
-	bool m_physics_override_sneak_glitch;
-	bool m_physics_override_new_move;
-	bool m_physics_override_sent;
+	float m_physics_override_speed = 1.0f;
+	float m_physics_override_jump = 1.0f;
+	float m_physics_override_gravity = 1.0f;
+	bool m_physics_override_sneak = true;
+	bool m_physics_override_sneak_glitch = false;
+	bool m_physics_override_new_move = true;
+	bool m_physics_override_sent = false;
```
Comment (SmallJoker) [inline src/content_sao.h]: Could you please line up these values for better readability?

File: src/guiEngine.h [hunk 6]
```diff
@@ -272,7 +272,7 @@ class GUIEngine {
 	void setTopleftText(const std::string &text);
 
 	/** pointer to gui element shown at topleft corner */
-	irr::gui::IGUIStaticText*	m_irr_toplefttext;
+	irr::gui::IGUIStaticText*	m_irr_toplefttext = nullptr;
```
Comment (SmallJoker) [inline src/guiEngine.h]: Pointer style wrong here too

File: src/content_sao.h (hunk 5 above)
Comment (nerzhul) [inline src/content_sao.h]: what do you mean ?

**Comment by Zeno-** [comment]
:+1:

File: src/collision.h [hunk 7]
```diff
@@ -36,37 +36,20 @@ enum CollisionType
 
 struct CollisionInfo
 {
-	enum CollisionType type;
-	v3s16 node_p; // COLLISION_NODE
-	bool bouncy;
+	CollisionInfo() {}
+	CollisionType type = COLLISION_NODE;
+	v3s16 node_p = v3s16(-32768,-32768,-32768); // COLLISION_NODE
```
Comment (Zeno-) [inline src/collision.h]: Is it safe to use -32768?

Comment (nerzhul) [inline src/collision.h]: the value fits in the container, i don't really know it's the original behaviour, i think we can keep this as is as it's not a PR which should trigger regression by changing values :)

**Pull request merged**